core/src/com/unciv/ui/tilegroups/TileGroup.kt

```diff
@@ -614,17 +614,21 @@ open class TileGroup(var tileInfo: TileInfo, var tileSetStrings:TileSetStrings)
             }
         }
     }
-
     fun updatePixelMilitaryUnit(showMilitaryUnit: Boolean) {
         var newImageLocation = ""
 
         val militaryUnit = tileInfo.militaryUnit
         if (militaryUnit != null && showMilitaryUnit) {
             val unitType = militaryUnit.type
-            val specificUnitIconLocation = tileSetStrings.unitsLocation + militaryUnit.name
```

Comment:
Since this exact combination is used 4 times in the code, it's a good idea to keep it

Reply:
modified to preserve this val as is. The new check seems like it ought to be efficient, not sure how to improve the performance there